mxcube3/core/components/user/usermanager.py

```diff
@@ -129,97 +149,104 @@ def signout(self):
 
             self.app.CURRENTLY_MOUNTED_SAMPLE = ""
 
-            self.db_set_in_control(flask_security.current_user, False)
+            self.db_set_in_control(flask_login.current_user, False)
 
             msg = "User %s signed out" % user.username
             logging.getLogger("MX3.HWR").info(msg)
 
-        flask.session.clear()
-        flask_security.logout_user()
+        # flask.session.clear()
+        flask_login.logout_user()
 
     def is_authenticated(self):
-        return flask_security.is_authenticated()
+        return flask_login.current_user.is_authenticated()
 
     def login_info(self):
-        # If user object has limsdata attribute if logged in:
-        if hasattr(flask_security.current_user, "limsdata"):
-            login_info = json.loads(flask_security.current_user.limsdata)
-        else:
-            login_info = {}
-
-        self.set_operator()
-
-        login_info = convert_to_dict(login_info)
-
-        proposal_list = [
-            {
-                "code": prop["Proposal"]["code"],
-                "number": prop["Proposal"]["number"],
-                "proposalId": prop["Proposal"]["proposalId"],
-            }
-            for prop in login_info["proposalList"]
-        ]
-
         res = {
             "synchrotronName": HWR.beamline.session.synchrotron_name,
             "beamlineName": HWR.beamline.session.beamline_name,
-            "loggedIn": True,
+            "loggedIn": False,
             "loginType": HWR.beamline.lims.loginType.title(),
-            "proposalList": proposal_list,
+            "proposalList": [],
             "user": {
-                "username": flask_security.current_user.username,
-                "email": flask_security.current_user.email,
-                "isstaff": "staff" in flask_security.current_user.roles,
-                "name": flask_security.current_user.name
-                if hasattr(flask_security.current_user, "name")
-                else "",
-                "inControl": flask_security.current_user.in_control,
-                "ip": flask_security.current_user.last_login_ip,
+                "username": "",
+                "email": "",
+                "isstaff": "",
+                "nickname": "",
+                "inControl": "",
+                "ip": "",
             },
         }
 
-        res["selectedProposal"] = "%s%s" % (
-            HWR.beamline.session.proposal_code,
-            HWR.beamline.session.proposal_number,
-        )
+        if not flask_login.current_user.is_anonymous:
+            login_info = convert_to_dict(json.loads(flask_login.current_user.limsdata))
+
+            self.update_operator()
+
+            proposal_list = [
```

Comment:
Hi @megyaz, I'm sorry I did not notice your comment until it was to late. You are maybe right that we need Person and Session I did not try the "select proposal" feature (we are not using it) so I cant really say. I think however that this data is already present on the Proposal structure itself. Maybe you can have a look at that ?. I agree that we otherwise need to add it in some good way.

Reply:
Thanks Marcus.
To be able to "select proposal", we need to add `Person` and `Session` here to `proposal_list`.